### PR TITLE
feat: changes for new desojs redirect flow

### PR DESCRIPTION
**Pull request by jackson-dean**
This is to integrate with the derived key identity lib https://github.com/deso-protocol/deso-workspace/pull/91

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -106,5 +106,9 @@
     "ts-node": "~8.3.0",
     "tslint": "~6.1.0",
     "typescript": "~4.4.0"
+  },
+  "volta": {
```
Comment (jackson-dean) [inline package.json:110]: this makes sure I'm running the correct versions of node and npm when I work on this. If you're not using this, it's pretty nice https://volta.sh

File: src/app/auth/google/google.component.ts [hunk 2]
```diff
@@ -227,6 +227,7 @@ export const getStateParamsFromGoogle = (
     deleteKey: false,
     derivedPublicKey: '',
     expirationDays: 0,
+    redirect_uri: '',
```
Comment (lazynina) [inline src/app/auth/google/google.component.ts:230]: we're also going to need to update `google-drive.service.ts` around line 97 to add `redirectUri: this.globalVars.redirectUri` so it gets fed into the google flow and we still have the param when we get back from the google flow.

File: src/app/log-in/log-in.component.html [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 <div
   class="container home-container text-center"
-  *ngIf="globalVars.inTab || globalVars.webview"
+  *ngIf="globalVars.inTab || globalVars.webview || globalVars.redirectURI"
```
Comment (lazynina) [inline src/app/log-in/log-in.component.html:5]: is there a reason we don't have `|| globalVars.callback` in here?

**Review by lazynina — CHANGES_REQUESTED** [review]

File: src/app/auth/google/google.component.ts (hunk 2 above)
Comment (jackson-dean) [reply src/app/auth/google/google.component.ts:230]: damn good catch, yeah i tried the google flow with popup but did not do the full page redirect + google flow. Just test and confirmed it was broken and this fixes it.

File: src/app/log-in/log-in.component.html (hunk 3 above)
Comment (jackson-dean) [reply src/app/log-in/log-in.component.html:5]: So my theory is that callback is really only used for mobile app use cases, and I believe mobile apps pass the webview param as well, so for the purpose of some of these template conditions I *think* they are interchangeable.

**Review by lazynina — APPROVED** [review]

**Pull request merged**